lib/chef/knife/windows_listener_create.rb

```diff
@@ -0,0 +1,106 @@
+        require "openssl" unless defined?(OpenSSL)
+      end
+
+      banner "knife windows listener create (options)"
+
+      option :cert_install,
+        short: "-c CERT_PATH",
+        long: "--cert-install CERT_PATH",
+        description: "Adds specified certificate to the Windows Certificate Store's Local Machine personal store before creating listener."
+
+      option :port,
+        short: "-p PORT",
+        long: "--port PORT",
+        description: "Specify port. Default is 5986",
+        default: "5986"
+
+      option :hostname,
+        short: "-h HOSTNAME",
+        long: "--hostname HOSTNAME",
+        description: "Hostname on the listener. Default is blank",
+        default: ""
+
+      option :cert_thumbprint,
+        short: "-t THUMBPRINT",
+        long: "--cert-thumbprint THUMBPRINT",
+        description: "Thumbprint of the certificate. Required only if --cert-install option is not used."
+
+      option :cert_passphrase,
+        short: "-cp PASSWORD",
+        long: "--cert-passphrase PASSWORD",
+        description: "Passphrase for certificate."
+
+      def get_cert_passphrase
+        config[:cert_passphrase] || ui.ask("Enter given certificate's passphrase (empty for no passphrase): ", echo: false)
+      end
+
+      def run
+        STDOUT.sync = STDERR.sync = true
+
+        unless ChefUtils.windows?
+          ui.error "WinRM listener can be created on Windows system only"
+          exit 1
+        end
+
+        begin
+          if config[:cert_install]
+            cert_passphrase = get_cert_passphrase
+            result = `powershell.exe -Command " '#{cert_passphrase}' | certutil  -importPFX '#{config[:cert_install]}' AT_KEYEXCHANGE"`
+            if $?.exitstatus
+              ui.info "Certificate installed to Certificate Store"
+              result = `powershell.exe -Command " echo (Get-PfxCertificate #{config[:cert_install]}).thumbprint "`
+              ui.info "Certificate Thumbprint: #{result}"
+              config[:cert_thumbprint] = result.strip
+            else
+              ui.error "Error installing certificate to Certificate Store"
+              ui.error result
+              exit 1
+            end
+          end
+
+          unless config[:cert_thumbprint]
+            ui.error "Please specify the --cert-thumbprint"
+            exit 1
+          end
+
+          result = `winrm create winrm/config/Listener?Address=*+Transport=HTTPS @{Hostname="#{config[:hostname]}";CertificateThumbprint="#{config[:cert_thumbprint]}";Port="#{config[:port]}"}`
+          Chef::Log.debug result
+
+          if $?.exitstatus == 0
+            ui.info "WinRM listener created with Port: #{config[:port]} and CertificateThumbprint: #{config[:cert_thumbprint]}"
+          else
+            ui.error "Error creating WinRM listener. use -VV for more details."
+            exit 1
+          end
+
+        rescue => e
+          puts "ERROR: + #{e}"
```

Comment:
This probably needs to be `ui.fatal` or something.

Reply:
Updated to ui.fatal!